packages/openapi3/test/status-codes.test.ts

```diff
@@ -0,0 +1,60 @@
+import { expectDiagnostics } from "@typespec/compiler/testing";
+import { deepStrictEqual } from "assert";
+import { diagnoseOpenApiFor, openApiFor } from "./test-host.js";
+
+describe("openapi3: response status codes", () => {
+  async function expectStatusCodes(code: string, statusCodes: string[]) {
+    const res = await openApiFor(code);
+    deepStrictEqual(Object.keys(res.paths["/"].get.responses), statusCodes);
+  }
+  it("map single status code", async () => {
+    await expectStatusCodes(
+      `
+      op read(): {@statusCode _: 200, content: string};
+      `,
+      ["200"]
+    );
+  });
+
+  it("map multiple status code", async () => {
+    await expectStatusCodes(
+      `
+      op read(): {@statusCode _: 200 | 201 | 204, content: string};
+      `,
+      ["200", "201", "204"]
+    );
+  });
+
+  it("map simple status code range", async () => {
+    await expectStatusCodes(
+      `
+      op read(): {@minValue(400) @maxValue(499) @statusCode _: int32, content: string};
```

Comment:
So if you set minValue 400 and maxValue 500 you will get an error?

Reply:
<img width="793" alt="image" src="https://github.com/microsoft/typespec/assets/1031227/a669f96c-41dd-4802-9011-81c518125757">
yes, should maybe just be a warning though